dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DeduplicationController.java

```diff
@@ -160,26 +160,44 @@ public PotentialDuplicate postPotentialDuplicate(
         return potentialDuplicate;
     }
 
-    @RequestMapping( method = { RequestMethod.PUT, RequestMethod.POST }, value = "/{id}/invalidation" )
-    public void markPotentialDuplicateInvalid(
-        @PathVariable String id )
+    @PutMapping( value = "/{id}" )
+    public void updatePotentialDuplicate( @PathVariable String id, @RequestParam( value = "status" ) String status )
         throws NotFoundException,
-        HttpStatusCodeException
+        HttpStatusCodeException,
+        BadRequestException
     {
+        checkDeduplicationStatusRequestParam( status );
+
         PotentialDuplicate potentialDuplicate = getPotentialDuplicateBy( id );
+        DeduplicationStatus deduplicationStatus = DeduplicationStatus.valueOf( status );
 
-        potentialDuplicate.setStatus( DeduplicationStatus.INVALID );
+        checkDbAndRequestStatus( potentialDuplicate, deduplicationStatus );
+
+        potentialDuplicate.setStatus( deduplicationStatus );
         deduplicationService.updatePotentialDuplicate( potentialDuplicate );
     }
 
+    private void checkDbAndRequestStatus( PotentialDuplicate potentialDuplicate,
+        DeduplicationStatus deduplicationStatus )
+        throws BadRequestException
+    {
+        if ( deduplicationStatus == DeduplicationStatus.MERGED )
+            throw new BadRequestException(
+                "Bad request. Can't update a potential duplicate to " + DeduplicationStatus.MERGED.name() );
+        if ( potentialDuplicate.getStatus() == DeduplicationStatus.MERGED )
+            throw new BadRequestException( "Bad request. Can't update a potential duplicate that is already "
+                + DeduplicationStatus.MERGED.name() );
+    }
+
     @DeleteMapping( value = "/{id}" )
     public void deletePotentialDuplicate(
         @PathVariable String id )
         throws NotFoundException,
         HttpStatusCodeException
     {
         PotentialDuplicate potentialDuplicate = getPotentialDuplicateBy( id );
-        deduplicationService.deletePotentialDuplicate( potentialDuplicate );
+        potentialDuplicate.setStatus( DeduplicationStatus.INVALID );
```

Comment:
Shouldn't have the same validation as the update method?
Does it make sense to mark as invalid a duplicate that was already on MERGED status?

Reply:
yes it makes sense, i'll ask @stian-sandvold to double check